### ISSUE TITLE
torch.optim.lr_scheduler

### ISSUE THREAD

**Issue by tejan-rgb** [pytorch/pytorch#54393]
## 📚 Documentation

`>>> # Assuming optimizer uses lr = 0.05 for all groups
>>> # lr = 0.05     if epoch < 30
>>> # lr = 0.005    if 30 <= epoch < 60
>>> # lr = 0.0005   if 60 <= epoch < 90
>>> # ...
>>> scheduler = StepLR(optimizer, step_size=30, gamma=0.1)
>>> for epoch in range(100):
>>>     train(...)
>>>     validate(...)
>>>     scheduler.step()`

The above is the code mentioned in the docs.
Now, if we see that at epoch == 30, the optimizer in the train function will use the initialized lr (i.e. 0.05) and after that when scheduler.step() command runs, it will update the lr, therefore the optimizer will use the updated lr at epoch == 31 and not at 30.
I have checked it out.

**Comment by kuraga** [comment]
Don't confirm on 2.2:
```python
optimizer = torch.optim.AdamW(model.parameters(), 0.05)
scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=30, gamma=0.1)
#scheduler = torch.optim.lr_scheduler.LinearLR(optimizer, start_factor=0.5, total_iters=4)
old_lr = None
for epoch in range(100):
    lr = scheduler.get_last_lr()[0]
    if lr != old_lr:
        print((epoch, lr))
        old_lr = lr
    scheduler.step()
```
```
(0, 0.05)
(30, 0.005)
(60, 0.0005)
(90, 0.00005)
```
